stan/math/prim/mat/matrix_addons.h

```diff
@@ -0,0 +1,5 @@
+#ifdef STAN_OPENCL
+#ifdef STAN_OPENCL_CACHE
```

Comment:
I think these ifdefs can go in prim/mat/fun/Eigen.hpp ?

Reply:
Yep!

Reply:
Just a note on why there are duplicate ifdefs here: 

There are bound to be other addons that will be added in this file so ifdefing the entire header wont work for long (@andrjohns will probably be the next to add stuff in the addons).

The other one in Eigen.hpp is there because there is no need to include an empty header file until we add other non-opencl related addons.